### PR TITLE
Fix bug

### PR DESCRIPTION
**Pull request by advpetc**
Modify ```_sink``` function.

### CODE REVIEW

**Review by billryan — COMMENTED** [review]
You should first check whether left/right index exceeds the length of array. I will review this issue this weekend.

P.S. Could you provide the test case for this bug?

File: zh-hans/basics_data_structure/heap.md [hunk 1]
```diff
@@ -37,9 +37,11 @@ class MaxHeap:
         # move node down the tree
         left, right = 2 * i + 1, 2 * i + 2
         max_index = i
-        if left < len(array) and array[left] > array[max_index]:
+        # should compare two chidren then determine which one to swap with
+        flag = array[left] > array[right]
```
Comment (billryan) [inline zh-hans/basics_data_structure/heap.md:41]: `left` may exceeds the length of array

Comment (advpetc) [reply zh-hans/basics_data_structure/heap.md:41]: yes, that's another condition needed to be considered.

**Comment by advpetc** [comment]
Also, note that it should have some way to overwrite the default comparator (or it won't support some customized instance comparison). I have completed code for MaxHeap written in C++, do you need me add/replace it here?

**Comment by billryan** [comment]
Great. You can add it here. 🍻

**Comment by billryan** [comment]
Thanks. 👍

**Pull request merged**